File: java/src/test/java/com/auth/ASAPAuthTest.java
package com.auth;

import com.auth.base.Authentication;
import com.auth.asap.ASAPAuth;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.http.RequestEntity;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

class ASAPAuthTest extends TestUtils {
    private Authentication auth;

    @BeforeEach
    void setUp() {
        super.setUp();
        auth = new ASAPAuth("test-issuer", "test-audience", "test-private-key");
    }

    @Test
    void testValidASAPToken() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer test-asap-token");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.OK, false);
    }

    @Test
    void testInvalidASAPToken() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer invalid-token");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testExpiredToken() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer expired-asap-token");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidAudience() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer invalid-audience-token");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidIssuer() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer invalid-issuer-token");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMissingAuthorizationHeader() {
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, null);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testMalformedAuthorizationHeader() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Bearer");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testInvalidScheme() {
        Map<String, String> headers = new HashMap<>();
        headers.put("Authorization", "Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ=");
        RequestEntity<Void> request = createTestRequest(HttpMethod.GET, headers);
        runAuthTest(auth, request, HttpStatus.UNAUTHORIZED, true);
    }

    @Test
    void testGetType() {
        assertAuthType(auth, "asap");
    }
} 

File: php/tests/ASAPAuthTest.php
<?php

namespace Tests\Auth;

use Auth\ASAPAuth;

class ASAPAuthTest extends TestUtils
{
    private $auth;

    protected function setUp(): void
    {
        parent::setUp();
        $this->auth = new ASAPAuth('test-issuer', 'test-audience', 'test-private-key');
    }

    public function testValidASAPToken(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer test-asap-token'
        ]);
        $this->runAuthTest($this->auth, $request, 200, false);
    }

    public function testInvalidASAPToken(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer invalid-asap-token'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMissingAuthorizationHeader(): void
    {
        $request = $this->createTestRequest('GET');
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testMalformedAuthorizationHeader(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Bearer'
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testInvalidScheme(): void
    {
        $request = $this->createTestRequest('GET', [
            'Authorization' => 'Basic dGVzdC11c2VyOnRlc3QtcGFzc3dvcmQ='
        ]);
        $this->runAuthTest($this->auth, $request, 401, true);
    }

    public function testGetType(): void
    {
        $this->assertAuthType($this->auth, 'asap');
    }
} 